CONTRIBUTING.md

```diff
@@ -19,53 +19,91 @@ Read the following guide if you're interested in contributing to cluster-api.
+All changes must be code reviewed. Coding conventions and standards are explained in the official [developer
+docs](https://github.com/kubernetes/community/tree/master/contributors/devel). Expect reviewers to request that you
+avoid common [go style mistakes](https://github.com/golang/go/wiki/CodeReviewComments) in your PRs.
 
 ## Backporting a Patch
 
-Cluster API ships older versions through `release-X.X` branches, usually backports are reserved to critical bug-fixes.
-Some release branches might ship with both Go modules and dep (e.g. `release-0.1`), users backporting patches should always make sure
-that the vendored Go modules dependencies match the Gopkg.lock and Gopkg.toml ones by running `dep ensure`
+Cluster API maintains older versions through `release-X.Y` branches. We accept backports of bug fixes to the most recent
+release branch. For example, if the most recent branch is `release-0.2`, and the `master` branch is under active
+development for v0.3.0, a bug fix that merged to `master` that also affects `v0.2.x` may be considered for backporting
+to `release-0.2`. We generally do not accept PRs against older release branches.
+
+## Breaking Changes
+
+Breaking changes are generally allowed in the `master` branch, as this is the branch used to develop the next minor
+release of Cluster API.
+
+Breaking changes are not allowed in release branches, as these represent minor versions that have already been released.
+These versions have consumers who expect the APIs, behaviors, etc. to remain stable during the life time of the patch
+stream for the minor release.
+
+Examples of breaking changes include:
+
+- Removing or renaming a field in a CRD
+- Removing or renaming a CRD
+- Removing or renaming an exported constant, variable, type, or function
+- Updating the version of critical libraries such as controller-runtime, client-go, apimachinery, etc.
+    - Some version updates may be acceptable, for picking up bug fixes, but maintainers must exercise caution when
+      reviewing.
+
+There may, at times, need to be exceptions where breaking changes are allowed in release branches. These are at the
+discretion of the project's maintainers, and must be carefully considered before merging. An example of an allowed
+breaking change might be a fix for a behavioral bug that was released in an initial minor version (such as `v0.3.0`).
 
 ### Merge Approval
 
-Cluster API maintainers may add "LGTM" (Looks Good To Me) or an equivalent comment to indicate that a PR is acceptable. Any change requires at least one LGTM.  No pull requests can be merged until at least one Cluster API maintainer signs off with an LGTM.
+Cluster API maintainers may add "LGTM" (Looks Good To Me) or an equivalent comment to indicate that a PR is acceptable.
+Any change requires at least one LGTM.  No pull requests can be merged until at least one Cluster API maintainer signs
```

Comment:
This is hard to do in practice, given that any k8s contributor can, in theory add /lgtm to anything

Reply:
This isn't net new but I can adjust it